blob/src/main/java/io/crate/blob/BlobContainer.java

```diff
@@ -91,21 +87,22 @@ private void createSubDirectories(File parentDir) {
         }
     }
 
-    public interface FileVisitor {
-
-        boolean visit(File file) throws IOException;
-
+    public Iterable<File> getFiles() {
+        final Collection<File> files = new LinkedList<>();
```

Comment:
Why is a LinkedList required here? I think it would be possible to implement a custom Iterable that does the traversal instead of adding everything into a list.